core/CronArchive.php

```diff
@@ -1008,15 +1034,20 @@ private function archiveReportsFor($idSite, $period, $date, $archiveSegments, Ti
                 return $success;
             }
 
-            $self = $this;
-            $request = new Request($url);
-            $request->before(function () use ($self, $url, $idSite, $period, $date) {
-                if ($self->isAlreadyArchivingUrl($url, $idSite, $period, $date)) {
-                     return Request::ABORT;
-                }
-            });
-            $urls[] = $request;
-            $this->logArchiveWebsite($idSite, $period, $date);
+            list($visits, $isThereArchive) = $this->isThereAValidArchiveForPeriod($idSite, $period, $segment);
+            if (!$isThereArchive) {
```

Comment:
wondering here as well similar to https://github.com/matomo-org/matomo/pull/14937/files?w=1#diff-c94ca2937035751f5e1ea942c529495dR1861 should we maybe move this into `1042` after ?
```php
 if ($self->isAlreadyArchivingUrl($url, $idSite, $period, $date)) {
                        return Request::ABORT;	                        return Request::ABORT;
                    }	                    }
```

Reply:
👍